frontstage/common/validators.py

```diff
@@ -0,0 +1,34 @@
+from wtforms.validators import InputRequired as IR
+from wtforms.validators import DataRequired as DR
+
+"""
+A collection of validators for use with WTForms forms
+"""
+
+
+class InputRequired(IR):
+    """
+    WTForms validator replacement that requires input, but doesn't add 'required' attribute to rendered html
+
+    Example sage: element = StringField('name', [ FrontstageInputRequired(message="Error message to show") ]
+    """
+    def __init__(self, message=None):
+        super(IR, self).__init__()
+        self.field_flags = remove_required(self.field_flags)
+
+
+class DataRequired(DR):
+    """
+    WTForms validator replacement that requires data, but doesn't add 'required' attribute to rendered html
+    """
+    def __init__(self, message=None):
+        super(DR, self).__init__()
+        self.field_flags = remove_required(self.field_flags)
+
+
+def remove_required(flags):
+    if 'required' in flags:
+        n = flags.index('required')
+        return flags[: n] + flags[n + 1:]
```

Comment:
Will this work if `required` is the last item in `flags`?

Reply:
Yeah, if you have a `List` and slice it with an out or range index as the `end` position, it corrects to max index instead.  This even works with an empty `List`.

See example below from Python REPL:
```
>>> l = []
>>> l2 = l[:3]
>>> l2
[]```

Two `List`s, (both alike in dignity) one empty, and one created from the out-of-bounds reference to a slice of the other.  Just returns what it can.

To address this concern further, I've added two extra tests that confirm that it is removed if it's either in the middle of, or at the end of, the list.

Happy to mark this resolved?

Reply:
Awesome :)